packages/make-styles/src/renderer/createDOMRenderer.ts

```diff
@@ -53,29 +104,30 @@ export function createDOMRenderer(targetDocument: Document = document): MakeStyl
           continue;
         }
 
-        const css = definition[1];
+        const bucketName = definition[0];
+        const css = definition[2];
         const ruleCSS = dir === 'rtl' ? rtlCSS || css : css;
 
-        renderer.insertionCache[cacheKey] = true;
+        const sheet = getStyleSheetForBucket(bucketName, target, renderer);
+
         try {
-          if (renderer.styleElement.sheet instanceof CSSStyleSheet) {
-            renderer.styleElement.sheet.insertRule(ruleCSS, renderer.index);
-            renderer.index++;
```

Comment:
We can't rely on `renderer.index` for `.insertRule()` as insertion runs for different style nodes now.